localstack-core/localstack/services/events/provider.py

```diff
@@ -511,6 +501,128 @@ def remove_targets(
         )
         return response
 
+    #########
+    # Archive
+    #########
+    @handler("CreateArchive")
+    def create_archive(
+        self,
+        context: RequestContext,
+        archive_name: ArchiveName,
+        event_source_arn: Arn,
+        description: ArchiveDescription = None,
+        event_pattern: EventPattern = None,
+        retention_days: RetentionDays = None,
+        **kwargs,
+    ) -> CreateArchiveResponse:
+        region = context.region
+        account_id = context.account_id
+        store = self.get_store(context)
```

Comment:
I am pretty sure that it should work with the context since no cross-region account communication can happen for archives. An archive can only be attached to a single event bus in the same region/account and replay events to only this event bus.